scripts/hgsc_subtypes_tybalt.py

```diff
@@ -0,0 +1,274 @@
+rnaseq_df = pd.read_table(rnaseq_file, index_col=0)
+rnaseq_df.shape
+
+
+# In[6]:
+
+ov_file = os.path.join('data', 'ov_subtype_info.tsv')
+ov_df = pd.read_table(ov_file, index_col=0)
+ov_df.head(2)
+
+
+# In[7]:
+
+encoded_file = os.path.join('data', "encoded_rnaseq_onehidden_warmup_batchnorm.tsv")
+encoded_df = pd.read_table(encoded_file, index_col=0)
+encoded_df.shape
+
+
+# In[8]:
+
+# Subset and merge the HGSC subtype info with the latent space feature activations
+ov_samples = list(set(encoded_df.index) & (set(ov_df.index)))
+
+ov_encoded = encoded_df.loc[ov_samples, ]
+ov_encoded_subtype = pd.merge(ov_df.loc[:, ['SUBTYPE', 'SILHOUETTE WIDTH']], ov_encoded,
+                              how='right', left_index=True, right_index=True)
+ov_encoded_subtype = ov_encoded_subtype.assign(subtype_color = ov_encoded_subtype['SUBTYPE'])
+
+ov_subtype_color_dict = {'Differentiated': 'purple',
+                         'Immunoreactive': 'green',
+                         'Mesenchymal': 'blue',
+                         'Proliferative': 'red'}
+ov_encoded_subtype = ov_encoded_subtype.replace({'subtype_color': ov_subtype_color_dict})
+
+print(ov_encoded_subtype.shape)
+ov_encoded_subtype.head(2)
+
+
+# In[9]:
+
+# Get the HGSC mean feature activation
+ov_mean_subtypes = ov_encoded_subtype.groupby('SUBTYPE').mean()
+ov_mean_subtypes
+
+
+# ## HGSC Subtype Math
+# 
+# Because of the relationship observed in the consistent clustering solutions, perform the following subtractions
+# 
+# 1. Immunoreactive - Mesenchymal
+# 2. Differentiated - Proliferative
+# 
+# The goal is to observe the features with the largest difference between the aformentioned comparisons. The differences should be in absolute directions
+
+# In[10]:
+
+mes_mean_vector = ov_mean_subtypes.loc['Mesenchymal', [str(x) for x in range(1, 101)]]
+imm_mean_vector = ov_mean_subtypes.loc['Immunoreactive', [str(x) for x in range(1, 101)]]
+pro_mean_vector = ov_mean_subtypes.loc['Proliferative', [str(x) for x in range(1, 101)]]
+dif_mean_vector = ov_mean_subtypes.loc['Differentiated', [str(x) for x in range(1, 101)]]
+
+
+# In[11]:
+
+high_immuno = (imm_mean_vector - mes_mean_vector).sort_values(ascending=False).head(2)
+high_mesenc = (imm_mean_vector - mes_mean_vector).sort_values(ascending=False).tail(2)
+
+print("Features with large differences: Immuno high, Mesenchymal low")
+print(high_immuno)
+print("Features with large differences: Mesenchymal high, Immuno low")
+print(high_mesenc)
+
+
+# In[12]:
+
+# Select to visualize encoding 56 because it has high immuno and low everything else
+ov_mean_subtypes.loc[:, ['87', '77', '56']]
+
+
+# In[13]:
+
+# Obtain the decoder weights
+weights = []
+for l in decoder.layers:
+    weights.append(l.get_weights())
+    
+weight_layer = pd.DataFrame(weights[1][0], columns=rnaseq_df.columns)
+weight_layer.head(2)
+
+
+# In[14]:
+
+def get_high_weight(weight_matrix, node, high_std=2, direction='positive'):
```

Comment:
Is this the best spot for this? I am not "pythonic" enough to know best. What do you think @danich1?

Reply:
It suffices since none of the variables below are being used outside the function call. Honestly, up to @gwaygenomics if he wants to leave it here or place it at the top. If this was a regular script, then I would say place this function at the time of the file.